addons/arsenalExport/functions/fnc_export.sqf

```diff
@@ -0,0 +1,106 @@
+            if (_forEachIndex == 0) then {_magCount = 1 max ((ceil (_wantedRounds / _magRounds)) - 2 * ((count _mags) - 1));};
+            _return = _return + format ['"%1:%2"', _x, _magCount];
+        };
+        if (_forEachIndex < ((count _mags) -1)) then {_return = _return + ", ";};
+    } forEach _mags;
+    if (_return == "") then {_return = '""';};
+    TRACE_4("getMags",_weapon,_mags,_wantedRounds,_return);
+    _return
+};
+
+private _lines = [];
+_lines pushBack format ["// Camo set"];
+_lines pushBack format ['#define CAMO_UNIFORM "%1"', GVAR(loadout_uniform)];
+_lines pushBack format ['#define CAMO_VEST "%1"', GVAR(loadout_vest)];
+_lines pushBack format ['#define CAMO_BACKPACK "%1"', GVAR(loadout_backpack)];
+_lines pushBack format ['#define CAMO_HEADGEAR "%1"', GVAR(loadout_headgear)];
+
+_lines pushBack format ["// Rifle"];
+_lines pushBack format ['#define RIFLE "%1"', GVAR(loadout_rifle)];
+_lines pushBack format ['#define RIFLE_MAG %1', [GVAR(loadout_rifle), GVAR(loadout_rifleMags), 300] call _fnc_getMags];
+_lines pushBack format ['#define RIFLE_ATTACHMENTS %1', [GVAR(loadout_rifleAttachments)] call _fnc_formatList];
+_lines pushBack format ['#define AAR_ATTACHMENTS RIFLE_ATTACHMENTS'];
+_lines pushBack format ['#define ALT_OPTICS "optic_Aco","rhsusf_acc_compm4","rhsusf_acc_eotech_xps3","CUP_optic_CompM2_Black","CUP_optic_TrijiconRx01_black","BWA3_optic_RSAS","RKSL_optic_RMR_MS19"'];
+
+_lines pushBack format ["// GL Rifle"];
+_lines pushBack format ['#define GLRIFLE "%1"', GVAR(loadout_glrifle)];
+_lines pushBack format ['#define GLRIFLE_MAG %1', [GVAR(loadout_glrifle), GVAR(loadout_rifleMags), 300] call _fnc_getMags];
+
+private _glMuzzle = (getArray (configFile >> "CfgWeapons" >> GVAR(loadout_glrifle) >> "muzzles")) param [1, "no2ndMuzzle"];
+private _glMags = getArray (configFile >> "CfgWeapons" >> GVAR(loadout_glrifle) >> _glMuzzle >> "magazines");
+switch (true) do {
+    case (({"1Rnd_Smoke_Grenade_shell" == _x} count _glMags) > 0):{
+        _lines pushBack format ['#define GLRIFLE_MAG_SMOKE "1Rnd_Smoke_Grenade_shell:2","1Rnd_SmokeRed_Grenade_shell:2"'];
+        _lines pushBack format ['#define GLRIFLE_MAG_HE "1Rnd_HE_Grenade_shell:5"'];
+        _lines pushBack format ['#define GLRIFLE_MAG_FLARE "UGL_FlareYellow_F:4"'];
+    };
+    case (({"rhs_GRD40_White" == _x} count _glMags) > 0):{
+        _lines pushBack format ['#define GLRIFLE_MAG_SMOKE "rhs_GRD40_White:2","rhs_GRD40_Red:2"'];
+        _lines pushBack format ['#define GLRIFLE_MAG_HE "rhs_VOG25:5"'];
+        _lines pushBack format ['#define GLRIFLE_MAG_FLARE "rhs_VG40OP_red:4"'];
+    };
+    default {
+        _lines pushBack format ["// WARNING - Unknown GL Muzzle [%1->%2->%3]", GVAR(loadout_glrifle), _glMuzzle, _glMags];
+        _lines pushBack format ['#define GLRIFLE_MAG_SMOKE "TODO:2","TODO:2"'];
+        _lines pushBack format ['#define GLRIFLE_MAG_HE "TODO:5"'];
+        _lines pushBack format ['#define GLRIFLE_MAG_FLARE "TODO:4"'];
+    };
+};
```

Comment:
Pretty sound logic, with the massCompat mod all RU should have access to GRD40s, all NATO should have the vanilla smokes